sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -2875,5 +2877,58 @@ await GetBlobClient(blobName)
                     .ConfigureAwait(false);
 
         #endregion DeleteBlob
+
+        #region GetBlobServiceClientCore
+        /// <summary>
+        /// Create a new <see cref="BlobServiceClient"/> that pointing to this <see cref="BlobContainerClient"/>'s blob service.
+        /// The new <see cref="BlobServiceClient"/>
+        /// uses the same request policy pipeline as the
+        /// <see cref="BlobContainerClient"/>.
+        /// </summary>
+        /// <returns>A new <see cref="BlobServiceClient"/> instance.</returns>
+        protected internal virtual BlobServiceClient GetBlobServiceClientCore()
+        {
+            BlobUriBuilder blobUriBuilder = new BlobUriBuilder(Uri)
+            {
+                // erase parameters unrelated to service
+                BlobContainerName = null,
+                BlobName = null,
+                VersionId = null,
+                Snapshot = null,
+            };
+
+            return new BlobServiceClient(
```

Comment:
Do we want to initialize a new client each time?  Or would we have a persistent Client as a private property?

Reply:
we can cache it since properties it's made don't change.